osu.Game/Screens/OnlinePlay/DailyChallenge/DailyChallengeIntro.cs

```diff
@@ -93,14 +93,15 @@ public DailyChallengeIntro(Room room)
         protected override BackgroundScreen CreateBackground() => new DailyChallengeIntroBackgroundScreen(colourProvider);
 
         [BackgroundDependencyLoader]
-        private void load(BeatmapDifficultyCache difficultyCache, BeatmapModelDownloader beatmapDownloader, OsuConfigManager config, AudioManager audio)
+        private void load(RulesetStore rulesets, BeatmapDifficultyCache difficultyCache, BeatmapModelDownloader beatmapDownloader, OsuConfigManager config, AudioManager audio)
         {
             const float horizontal_info_size = 500f;
 
-            Ruleset ruleset = Ruleset.Value.CreateInstance();
-
             StarRatingDisplay starRatingDisplay;
 
+            IBeatmapInfo beatmap = item.Beatmap;
+            Ruleset ruleset = rulesets.GetRuleset(item.Beatmap.Ruleset.ShortName)!.CreateInstance();
```

Comment:
Have you actually tested this in practice? I'm not sure that the API is even giving us a `ShortName`. Following the logic:

`PlaylistItem.apiBeatmap`, used for deserialisation, is of type `APIBeatmap`

https://github.com/ppy/osu/blob/321e509f11416071588a531dc8021667f7fff834/osu.Game/Online/Rooms/PlaylistItem.cs#L46-L56

`APIBeatmap.Ruleset` only sets the `OnlineID`:

https://github.com/ppy/osu/blob/321e509f11416071588a531dc8021667f7fff834/osu.Game/Online/API/Requests/Responses/APIBeatmap.cs#L122

Reply:
Ah, there was one missing piece to the puzzle...

`APIRuleset.ShortName`:

https://github.com/ppy/osu/blob/321e509f11416071588a531dc8021667f7fff834/osu.Game/Online/API/Requests/Responses/APIBeatmap.cs#L137-L155

Reply:
I have tested yes.